pkg/log/log.go

```diff
@@ -1,23 +1,56 @@
 package log
 
 import (
+	"fmt"
 	"strings"
 
 	"github.com/sirupsen/logrus"
 )
 
-type RemoveSecretFormatterDecorator struct {
+//PiperLogFormatter is the custom formatter of piper
+type PiperLogFormatter struct {
 	logrus.TextFormatter
+	plainFormat         bool
+	defaultFormat       bool
+	withTimeStampFormat bool
+	fullFormat          bool
```

Comment:
These are exclusive, so a single field would simplify the code a lot. It would even work to directly store the string passed as logFormat here, and then have a switch() statement with default case in the Format() function. (You can have strings as case expression.) This wouldn't change the code functionality at all, and make it easy to extend for a new log format in just a single place (by adding a case).

Reply:
agreed, updated, thanks